vignettes/decorate-module-output.Rmd

```diff
@@ -0,0 +1,467 @@
+---
+title: "Customizing Module Output"
+author: "NEST CoreDev"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Customizing Module Output}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`teal` is a powerful `shiny`-based framework with built-in modules for interactive data analysis. 
+This document outlines the customization options available for modifying the output of `teal` modules. 
+You will learn how to use [`teal_transform_module`](https://insightsengineering.github.io/teal/latest-tag/reference/teal_transform_module.html) to modify and enhance the objects created by `teal::modules()`, 
+enabling you to tailor the outputs to your specific requirements without rewriting the original module code.
+
+## Decorators
+
+One of ways of adjusting input data or customizing module outputs in `teal` is the usage of `transformators` 
+created through `teal_transform_module`. This document will focus on modifying outputs, and for clarity, we will refer 
+to these output-modifying objects as decoration objects or decorators.
+
+In below chapter we will present how to create the simplest static decorator with just a server part. Later, we will
+present examples on more advanced usage, where decorators contain UI. You will also learn about a convenience 
+function that makes it easier to write decorators, called [`make_teal_transform_server`](https://insightsengineering.github.io/teal/latest-tagx/reference/make_teal_transform_server.html). The chapter ends with an 
+example module that utilizes decorators and a snippet that uses this module in `teal` application.
```

Comment:
Let's introduce the concept of decoration - this is still not covered and I think it's important to introduce it first. Something like below - feel free to edit.

Let's assume that there is an object `x` of class `"my_class"`. _Decoration_ is a process of changing an object `x` while preserving its class. These changes might add new methods, change data, etc. In simple terms, `foo(x)` is a decoration of `x` if `foo()` preserve input object class. We name `x` as "decorated object" and `foo()` as "decoration function".
In the context of `teal` applications, we always refer to decorating module outputs - plots, tables, etc. In such case, a simple example could be `add_title(x, <plot title>)` function that takes `ggplot2` plot object as `x` argument, apply title change(s) and returns `ggplot2` plot object. It's important to preserve the class so that the "display part" that follows can be run on both decorated and not decorated objects.
The decoration process might be relatively simple (e.g. a single call without additional data) or very complex (e.g. multiple interdependent steps - each requiring additional data coming from dedicated shiny). This functionality is designed to be a power tool for application developers allowing significant output customisation that are not achievable through existing set of module parameters.

Reply:
Thanks @pawelru 
I redacted the text and included as such f329c0eab09a58ad07e9dcf5205377980eeb8019